PowerShell/Azure/Storage/Invoke-AzureWebAppStorageAzCopy/Invoke-AzureWebAppStorageAzCopy.psm1

```diff
@@ -0,0 +1,354 @@
+                        ErrorAction = 'Stop'
+                    }
+                    New-AzStorageContainer @newContainerParameters
+                }
+
+                # Requesting access token for the destination container and constructing the copy URL.
+                $destinationAccessTokenParameters = @{
+                    Context = $destinationStorageContext
+                    Permission = 'lrwd'
+                    ExpiryTime = (Get-Date).AddMinutes($SasLifetimeMinutes)
+                }
+                $destinationAccessToken = New-AzStorageAccountSASToken @accessTokenCommonParameters @destinationAccessTokenParameters
+                if ($destinationAccessToken -notlike '?*') { $destinationAccessToken = "?$destinationAccessToken" }
+                $currentContainerDestination = "https://$($destinationStorageConnection.AccountName).blob.core.windows.net/$($destinationContainerName + $destinationAccessToken)"
+            }
+            # If the destination path is overriden and is a valid local path, then create the destination folder for the
+            # container, if it doesn't exist yet.
+            elseif ($DestinationAbsolutePath -and [System.IO.Path]::IsPathRooted($DestinationAbsolutePath) -and (Test-Path $DestinationAbsolutePath -IsValid))
+            {
+                $currentContainerDestination = Join-Path $DestinationAbsolutePath $sourceContainer.Name
+
+                if (-not (Test-Path $currentContainerDestination -PathType Container))
+                {
+                    New-Item -Path $currentContainerDestination -ItemType Directory
+                }
+            }
+            else
+            {
+                throw "'$DestinationAbsolutePath' is not a valid local path for DestinationAbsolutePath!"
+            }
+
+            # Requesting access token for the source container and constructing the copy URL.
+            $sourceAccessTokenParameters = @{
+                Context = $sourceStorageContext
+                Permission = 'lr'
+                ExpiryTime = (Get-Date).AddMinutes($SasLifetimeMinutes)
+            }
+            $sourceAccessToken = New-AzStorageAccountSASToken @accessTokenCommonParameters @sourceAccessTokenParameters
+            if ($sourceAccessToken -notlike '?*') { $sourceAccessToken = "?$sourceAccessToken" }
+            $sourceContainerUrl = "https://$($sourceStorageConnection.AccountName).blob.core.windows.net/$($sourceContainer.Name + $sourceAccessToken)"
+
+            # And finally, the actual copy operation.
+            # WARNING: The first two unnamed parameters are the source and the destination in this order.
+            azcopy sync $sourceContainerUrl $currentContainerDestination --recursive=true --s2s-preserve-access-tier=$preserveAccessTierOnDestination --delete-destination=$RemoveExtraFilesOnDestination --include-regex="$($IncludePathRegexes -join ';')" --exclude-regex="$($ExcludePathRegexes -join ';')"
```

Comment:
Very long line.

Reply:
Fixed, but I had a super annoying behavior which completely derailed an otherwise easy fix: When constructing the parameter array, surrounding the regex strings with apostrophes or double quotes (e.g., `--include-regex="^GitHgMirror\/"`) made them completely ineffective. Even though its string representation was exactly the same as what was working before when running the command as plain string. The upside is that I did (= had to) more local testing. :)

Reply:
/sad :(.